packages/flutter/lib/src/material/calendar_date_picker.dart

```diff
@@ -1008,7 +1013,7 @@ class _DayPickerState extends State<_DayPicker> {
             focusNode: _dayFocusNodes[day - 1],
             onTap: () => widget.onChanged(dayToBuild),
             radius: _dayPickerRowHeight / 2 + 4,
-            statesController: MaterialStatesController(states),
+            statesController: _materialStatesController,
```

Comment:
This doesn't look right, every date widget should have a separate states controller. cc @HansMuller

Reply:
That's right. each day widget should have its own selected/disabled/etc state.

Reply:
It is interesting that tests passed...

Reply:
Fixed. Looks better?

Reply:
Yes that looks better although factoring out a stateful day widget per @chunhtai suggestion seems like a cleaner way to do this. 

And, [as you pointed out](https://github.com/flutter/flutter/pull/133884#discussion_r1319168307), it looks like need some additional tests.

Reply:
Added test.

I expect creation of separate widget for a day will take 2-3 more iterations, while I want to use the time to test cover other tests for leaks. 
Added TODO about it.

Reply:
I expect separate widget for a day will take 2-3 iterations, while I prefer to prioritize test coverage. Created TODO: https://github.com/flutter/flutter/issues/134323.